src/media/inner.rs

```diff
@@ -1241,6 +1354,45 @@ impl MediaInner {
     fn has_tx_rtx(&self) -> bool {
         self.sources_tx.iter().any(|s| s.is_rtx())
     }
+
+    fn padding_snapshot(&self, now: Instant) -> QueueSnapshot {
+        let (padding_count, padding_size, padding_queue_time, first_unsent) = self
+            .padding
+            .iter()
+            .fold((0_u32, 0, Duration::ZERO, None), |acc, p| {
+                let is_first = acc.1 == 0;
+                match p {
+                    Padding::Blank {
+                        requested_at, size, ..
+                    } => (
+                        acc.0 + 1,
+                        acc.1 + size,
+                        acc.2 + now.duration_since(*requested_at),
+                        if is_first { Some(*requested_at) } else { acc.3 },
+                    ),
+                    Padding::Spurious(Resend {
+                        body_size,
+                        queued_at,
+                        ..
+                    }) => (
+                        acc.0 + 1,
+                        acc.1 + body_size,
+                        acc.2 + now.duration_since(*queued_at),
+                        if is_first { Some(*queued_at) } else { acc.3 },
+                    ),
+                }
+            });
```

Comment:
Wouldn't it be cleaner to do `impl From<Padding> for QueueSnapshot` and use the `merge` function we already have for snapshot?

Reply:
Can't really do `From` but I did something similar to the resends

Reply:
See https://github.com/algesten/str0m/pull/140/commits/9f7fc94ca6fa60c9bda01ff07729e6dec5ecc1e8